### DISCUSSION TITLE
V1.0 rename some functions?

### DISCUSSION THREAD

**Discussion by duongphuhiep** [firasdarwish/ore#10] [General]
Some ore's functions can have a better name, I will list my suggestion + reasoning. It's pretty subjectif. So it is fine if the suggestion will be rejected.

* `RegisterEagerSingleton` => `RegisterSingletonValue`: "Value" is more obvious than "Eager"
* `RegisterLazyCreator` => `RegisterCreator`: the word "Lazy" can appeare in the godoc of this function and it's enough
* `RegisterLazyFunc` => `RegisterFunc`:  the word "Lazy" can appeare in the godoc of this function. Moreover evens without godoc, when passing a func as parameter, everybody would expect that it is "lazy".
* `Build()` => `SealRegistration`: it is not obvious to understand that `Build()` won't actually build anything but is used to seal further registrations.

**Comment by firasdarwish** [comment]
i've renamed `Build()` to `Seal()` in my latest PR

this one is pretty related to #11 for the naming concerns as well

i totally agree with you, the naming i used is way too explicit i could imagine myself a Java developer ^_^

as for methods suffixed with `ToContainer`, do you think we could also do something about it ?
since the current implementation makes the container very self-sufficient , i would see my self suggesting remove `ToContainer` methods and replacing them to methods invoked on the container itself

```go
con := ore.NewContainer()

con.RegisterCreator(....)
```

what do you think ?

**Comment by duongphuhiep** [comment]
I don't like the `ToContainer` suffix either, but it is the Go limitation that you can't have the method Generic. This code won't compile

```go
func (this *Container) RegisterCreator[T any](...) {}
```

so we don't have choice

```go
RegisterCreatorToContainer[T any](this *Container, ...) {}
```
